### DISCUSSION TITLE
Support for system file browser when selecting files

### DISCUSSION THREAD

**Discussion by 25pwn** [ppy/osu#16326] [Ideas] [answered]
The 'Import files' and 'Change folder location...' selector does not support showing hidden files, manually typing the location etc. Support for using something else like [kdialog](https://develop.kde.org/deploy/kdialog/#file-selection-dialogs) or [zenity](https://help.gnome.org/users/zenity/stable/file-selection.html.en) instead would be appreciated.

**Comment by peppy** [comment, marked as answer]
These will not ever be supported. Instead,

- Hidden file support: https://github.com/ppy/osu/discussions/15759
- Typing location can also be added is it is asked for by enough users.

**Reply by 25pwn** [reply]
Not ever, as in a PR will not be accepted?

**Reply by peppy** [reply]
Correct. We aren't planning on supporting any per-platform modal dialogs unless the system requires it.
